### PR TITLE
Recipe names follow PyPI naming (only hyphens, no underscores)

### PR DESCRIPTION
**Pull request by ForgottenProgramme**
Added a 'canonicalize_name' method to change all underscores to hyphen in the recipe name. 

user input: ``grayskull pypi slack_sdk``
Recipe generated: ``slack-sdk``

Solves #228 

See this Gitter conversation: [June 18, 2021 2:03 PM](https://gitter.im/conda_grayskull/community?at=60cc5a48d20143617e504c2e)

22/06/2021
Edit:

- We want to decide how recipes generated via grayskull are named.
- Presently grayskull names the recipes (after finding the project on PyPI) same as what is entered by the user. ``grayskull pypi <project_name_entered_by_user>``
- The alternatives: 
1. Use the PyPI name from metadata
2. Use the 'original' project name from metadata

### CODE REVIEW

**Comment by ForgottenProgramme** [comment]
slack_sdk

The messages displayed still show 'slack_sdk'. Maybe that needs to be changed as well? 
I'll make a different PR for that.

**Comment by ForgottenProgramme** [comment]
@marcelotrevisani

File: grayskull/pypi/pypi.py [hunk 1]
```diff
@@ -55,9 +55,15 @@ def __init__(
...
         self._setup_metadata = None
         self._is_arch = False
+        name = self._canonicalize_name(name)
         super(PyPi, self).__init__(name=name, version=str(version) if version else None)
         self["build"]["script"] = "<{ PYTHON }} -m pip install . -vv"
 
+    @staticmethod
+    def _canonicalize_name(name:str ):
+        """replace all underscores with hyphens in package name"""
+        return name.replace("_","-")
```
Comment (ocefpaf) [inline grayskull/pypi/pypi.py:65]: While most projects use `-` in the pkg name and `_` in the import name that is not always true. I'm not sure we have enough metadata to figure that out automatically, if so, we should. If not, the implementation here is making an assumption that may, in some rare cases, fail.

TL;DR I'm not against it but I would prefer a metadata based approach first and a fallback to something we all agree on.

Comment (marcelotrevisani) [reply grayskull/pypi/pypi.py:65]: I agree with @ocefpaf 

we recover the project name from ``setup.py`` and pypi metadata and after it grayskull try to merge both information, right now it discard the name and does not set it, one way to do it would be:

in the method ``_extract_requirements`` inside of the ``PyPi`` class online [781](https://github.com/conda-incubator/grayskull/blob/2e2ab84beb8dbb71a23eb43abcecb5b48eba949a/grayskull/pypi/pypi.py#L781) we have:
```python
        name = metadata["name"]
```
so, one thing which we can do is:
```python
        name = metadata["name"]
        self.set_jinja_var("name", name)
```

that should solve it for now

**Comment by marcelotrevisani** [comment]
@ocefpaf , this project ``slack_sdk`` has different names on pypi and on project
pypi metadata is showing as ``slack-sdk`` https://pypi.org/pypi/slack-sdk/json
however, the project name is: ``slack_sdk`` https://github.com/slackapi/python-slack-sdk/blob/148fbb181a2ba314518118aaa8e590d17ae3eb10/setup.py#L250

which one should we follow? I think it would be better to follow the project name instead of pypi, because you know how problematic pypi is...

**Comment by ForgottenProgramme** [comment]
@ocefpaf 
Hello, Filipe!
Would you please add your comments on this issue in this grayskull gitter chat? https://gitter.im/conda_grayskull/community
It would be a lot easier to discuss there.

**Comment by ocefpaf** [comment]
We are hitting the issues where PyPI doesn't care about _ vs - but we do because conda isn't python only. What we did in the past, for cases like these, was to use multiple outputs to provide both names. However, that is not an official practice. 

Some people dislike having an empty pkg just to reserve a name.

TL;DR let's pause this and discuss with more core members to see if we can agree on something.

**Comment by maresb** [comment]
I was just looking at some old issues, and I noticed that this was closed. What was the conclusion in the end?
